### ISSUE TITLE
Emmet custom snippets no longer working

### ISSUE THREAD

**Issue by johnrobertporter** [Microsoft/vscode#33210]
- VSCode Version: 1.15.1
- OS Version: OS X 10.12.6

Since the release of Emmet 2.0, my custom snippets no longer work. As a pared down example:

```json
{
	"xml": {
		"snippets": {
			"xml": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
		}
	}
}
```

Using the above, the snippet doesn't work. It worked perfectly alongside many others pre Emmet 2.0.

Steps to Reproduce:

1. Install the custom emmet `snippets.json` file, and declare that folder location in the settings
2. Set new file language to xml.
3. Use snippet `xml`, nothing happens.
4. Set Emmet to use tab for auto completion.
5. Use snippet `xml` and press tab, nothing happens.

Reproduces without extensions: Yes

The release of Emmet 2.0 has severely ruined my productivity and my opinion is that the changes implemented were a very bad idea without much more user feedback. You changed a system that has worked fine in many many editors for nearly a decade.

**Comment by user3323** [comment]
Yeah, I think that's because Emmet doesn't search on properties now, but only on values

I wrote about it here - https://github.com/Microsoft/vscode/issues/32911#issuecomment-324806560

I really don't understand why dev team doesn't want to fix this obnoxious behavior. Maybe they wait some hundreds of users will ask them about fix this problem. And now here is only couple of us.

**Comment by ramya-rao-a** [comment]
@user3323 

> Yeah, I think that's because Emmet doesn't search on properties now, but only on values

No, that statement and your comment https://github.com/Microsoft/vscode/issues/32911#issuecomment-324806560 are related to css snippets and triggering of css emmet suggestions. This issue is about non css snippets.

No, I am not waiting for hundreds of users to ask me or file issues. I know #32911  is an issue and am working on it. 

@johnrobertporter

> the snippet doesn't work. It worked perfectly alongside many others pre Emmet 2.0.

 In Emmet 2.0, custom snippets for markup (html/xml/xsl/jade etc) need to be valid abbreviations themselves. If the snippet cannot be represented by an abbreviation, then you can wrap them inside `{}` and they will get used as is. You can read about it in the [Emmet docs on custom snippets in VS Code](https://code.visualstudio.com/docs/editor/emmet#_using-custom-emmet-snippets) or in the [Release Notes for 1.15](https://code.visualstudio.com/updates/v1_15#_emmet-20)

This is a design decision by the [Sergey Chikuyonok](https://github.com/sergeche), author of Emmet itself to make snippets work in his new [modular approach taken for Emmet 2.0](https://code.visualstudio.com/blogs/2017/08/07/emmet-2.0#_new-modular-approach-to-emmet). 

So in your case, it would be
```json
{
	"xml": {
		"snippets": {
			"xml": "{<?xml version=\"1.0\" encoding=\"UTF-8\"?>}"
		}
	}
}
```

> Emmet 2.0 has severely ruined my productivity 

Is there anything else other than the custom snippets that is not working as expected?

**Comment by johnrobertporter** [comment]
As I say, I have a lot of snippets, so if they can't work in Emmet 2.0, then that is my productivity gone.

I will try as you suggest and report back here. I didn't realise the change was from Emmet itself, as the website documentation still talks about both approaches (https://docs.emmet.io/customization/snippets/)

**Comment by user3323** [comment]
@ramya-rao-a 

> No, I am not waiting for hundreds of users to ask me or file issues. I know #32911 is an issue and am working on it.

Thanks for your work.
I really don't understand why do we need CSS snippets in snippets.json (_hello, ironic tautology_), if Emmet doesn't count them? If it counts only matching letters in values...

So, let me copy snippet from that topic:
```
  "css": {
    "snippets": {
      "xx": "xaaaaaa"
    }
  }
```
It is simple as hell.
I have snippet called "xx".
I type "xx" - and I have nothing in the snippets list. How? Why?
I think it's totally wrong and that's what I call obnoxious behavior.

**Comment by ramya-rao-a** [comment]
@johnrobertporter I figured out one way to ease the pain of converting the custom snippets to the new format.

Which is to wrap the snippet value inside `{}` in code if I find any snippet value that starts with `<` and ends with `>`. 

This should enable to use all your old style snippets in Emmet 2.0

(Don't know why I didn't think of this earlier...)

I have pushed in the change, and the VS Code Insiders that will be out in a few hours will have the change.

The only catch is that if you had a snippet say `<ul>hello</ul>` for `html` and you expected it to work in say `jade`, you will get `<ul>hello</ul>` instead of `ul hello`

@user3323 I'll get back to your css snippet concerns in https://github.com/Microsoft/vscode/issues/32911
